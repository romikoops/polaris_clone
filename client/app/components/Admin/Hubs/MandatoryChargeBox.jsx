import React, { Component } from 'react'
import { withNamespaces } from 'react-i18next'
import styles from './index.scss'
import PropTypes from '../../../prop-types'
import AdminPromptConfirm from '../Prompt/Confirm'
import Checkbox from '../../Checkbox/Checkbox'
import { gradientTextGenerator } from '../../../helpers'

class MandatoryChargeBox extends Component {
  constructor (props) {
    super(props)
    this.state = {
      mandatoryCharge: {}
    }
  }
  componentWillReceiveProps (nextProps) {
    if (nextProps.mandatoryCharge && nextProps.mandatoryCharge.id !== this.state.mandatoryCharge.id) {
      this.setState({ mandatoryCharge: nextProps.mandatoryCharge })
    }
  }
  confirmDelete () {
    this.setState({
      confirm: true
    })
  }
  closeConfirm () {
    this.setState({ confirm: false })
  }
  confirmSave (target) {
    this.setState({ confirm: true })
  }
  closeAndSave () {
    const { saveChanges } = this.props
    const { mandatoryCharge } = this.state
    saveChanges(mandatoryCharge)
    this.closeConfirm()
  }
  handleToggle (ev, key) {
    this.setState({
      mandatoryCharge: {
        ...this.state.mandatoryCharge,
        [key]: !this.state.mandatoryCharge[key]
      }
    })
  }

  render () {
    const { mandatoryCharge, confirm } = this.state
    const {
      theme, t
    } = this.props
    if (!mandatoryCharge.id) return ''

    const confimPrompt = confirm ? (
      <AdminPromptConfirm
        theme={theme}
        heading={t('common:areYouSure')}
        text={t('admin:instantlyAvailable')}
        confirm={() => this.closeAndSave()}
        deny={() => this.closeConfirm()}
      />
    ) : (
      ''
    )
    const iconTheme =
      theme && theme.colors
        ? gradientTextGenerator(theme.colors.primary, theme.colors.secondary)
        : 'black'
    const editHasOccureed = mandatoryCharge.import_charges !== this.props.mandatoryCharge.import_charges ||
    mandatoryCharge.export_charges !== this.props.mandatoryCharge.export_charges

    return (
      <div className="flex-100 layout-row layout-align-start-start layout-wrap">
        { confimPrompt }
        <div className={`flex-100 layout-row layout-align-start-center layout-wrap ${styles.mandatory_charges_box}`}>
          <div className={`flex-75 layout-row layout-align-space-around-center ${styles.charges_row_padding}`}>
            <p className="flex-none">{t('admin:importFees')}</p>
            <Checkbox
              id="import"
              theme={theme}
              name="import"
              checked={mandatoryCharge.import_charges}
              onChange={e => this.handleToggle(e, 'import_charges')}
            />
          </div>
          <div className={`flex-75 layout-row layout-align-space-around-center ${styles.charges_row_padding}`}>
            <p className="flex-none">{t('admin:exportFees')}</p>
            <Checkbox
              id="export"
              theme={theme}
              name="export"
              checked={mandatoryCharge.export_charges}
              onChange={e => this.handleToggle(e, 'export_charges')}
            />
          </div>
        </div>
        <div className="flex-100 layout-row layout-align-end-center">
          {editHasOccureed ? (
            <div className={`${styles.save_icon_btn} flex-none layout-row layout-align-end-center pointy`} onClick={() => this.confirmSave()}>
              <p className="flex-none">{t('admin:save')}</p>
              <div className={`${styles.save_icon_btn} flex-none layout-row pointy`}>
                <i className="fa fa-floppy-o clip" style={iconTheme} />
              </div>
            </div>
          ) : (
            ''
          )}
        </div>
      </div>
    )
  }
}

MandatoryChargeBox.propTypes = {
  t: PropTypes.func.isRequired,
  theme: PropTypes.theme.isRequired,
  saveChanges: PropTypes.func.isRequired,
  mandatoryCharge: PropTypes.func.isRequired
}
MandatoryChargeBox.defaultProps = {}
export default withNamespaces(['admin', 'common]'])(MandatoryChargeBox)
