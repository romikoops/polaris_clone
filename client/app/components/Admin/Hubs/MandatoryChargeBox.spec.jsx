import * as React from 'react'
import { shallow } from 'enzyme'
import { theme, identity } from '../../../mock'

import MandatoryChargeBox from './MandatoryChargeBox'

const propsBase = {
  theme,
  saveChanges: identity,
  mandatoryCharge: identity
}

test('shallow render', () => {
  expect(shallow(<MandatoryChargeBox {...propsBase} />)).toMatchSnapshot()
})

test('theme is falsy', () => {
  const props = {
    ...propsBase,
    theme: null
  }
  expect(shallow(<MandatoryChargeBox {...props} />)).toMatchSnapshot()
})

test('state.mandatoryChange.id is truthy', () => {
  const wrapper = shallow(<MandatoryChargeBox {...propsBase} />)
  wrapper.setState({ mandatoryCharge: { id: 1 } })

  expect(wrapper).toMatchSnapshot()
})

test('editHasOccured is true', () => {
  const wrapper = shallow(<MandatoryChargeBox {...propsBase} />)

  wrapper.setState({
    mandatoryCharge: {
      id: 1,
      import_charges: 'IMPORT_CHARGES'
    }
  })

  expect(wrapper).toMatchSnapshot()
})

test('state.confirm is truthy', () => {
  const wrapper = shallow(<MandatoryChargeBox {...propsBase} />)

  wrapper.setState({
    mandatoryCharge: { id: 1 },
    confirm: true
  })

  expect(wrapper).toMatchSnapshot()
})
