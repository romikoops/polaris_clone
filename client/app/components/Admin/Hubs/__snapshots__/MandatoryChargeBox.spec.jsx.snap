// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editHasOccured is true 1`] = `
<div
  className="flex-100 layout-row layout-align-start-start layout-wrap"
>
  <div
    className="flex-100 layout-row layout-align-start-center layout-wrap mandatory_charges_box"
  >
    <div
      className="flex-75 layout-row layout-align-space-around-center charges_row_padding"
    >
      <p
        className="flex-none"
      >
        Import Fees
      </p>
      <Checkbox
        checked="IMPORT_CHARGES"
        className=""
        disabled={false}
        id="import"
        name="import"
        onChange={[Function]}
        onClick={null}
        size={null}
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <div
      className="flex-75 layout-row layout-align-space-around-center charges_row_padding"
    >
      <p
        className="flex-none"
      >
        Export Fees
      </p>
      <Checkbox
        checked={false}
        className=""
        disabled={false}
        id="export"
        name="export"
        onChange={[Function]}
        onClick={null}
        size={null}
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
  </div>
  <div
    className="flex-100 layout-row layout-align-end-center"
  >
    <div
      className="save_icon_btn flex-none layout-row layout-align-end-center pointy"
      onClick={[Function]}
    >
      <p
        className="flex-none"
      >
        Save
      </p>
      <div
        className="save_icon_btn flex-none layout-row pointy"
      >
        <i
          className="fa fa-floppy-o clip"
          style={
            Object {
              "background": "-webkit-linear-gradient(-90deg, #333,#fafafa)",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`shallow render 1`] = `""`;

exports[`state.confirm is truthy 1`] = `
<div
  className="flex-100 layout-row layout-align-start-start layout-wrap"
>
  <AdminPromptConfirm
    confirm={[Function]}
    deny={[Function]}
    heading="Are you sure?"
    t={[Function]}
    text="These changes will be instantly available in your store"
    theme={
      Object {
        "colors": Object {
          "primary": "#333",
          "secondary": "#fafafa",
        },
      }
    }
  />
  <div
    className="flex-100 layout-row layout-align-start-center layout-wrap mandatory_charges_box"
  >
    <div
      className="flex-75 layout-row layout-align-space-around-center charges_row_padding"
    >
      <p
        className="flex-none"
      >
        Import Fees
      </p>
      <Checkbox
        checked={false}
        className=""
        disabled={false}
        id="import"
        name="import"
        onChange={[Function]}
        onClick={null}
        size={null}
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <div
      className="flex-75 layout-row layout-align-space-around-center charges_row_padding"
    >
      <p
        className="flex-none"
      >
        Export Fees
      </p>
      <Checkbox
        checked={false}
        className=""
        disabled={false}
        id="export"
        name="export"
        onChange={[Function]}
        onClick={null}
        size={null}
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
  </div>
  <div
    className="flex-100 layout-row layout-align-end-center"
  />
</div>
`;

exports[`state.mandatoryChange.id is truthy 1`] = `
<div
  className="flex-100 layout-row layout-align-start-start layout-wrap"
>
  <div
    className="flex-100 layout-row layout-align-start-center layout-wrap mandatory_charges_box"
  >
    <div
      className="flex-75 layout-row layout-align-space-around-center charges_row_padding"
    >
      <p
        className="flex-none"
      >
        Import Fees
      </p>
      <Checkbox
        checked={false}
        className=""
        disabled={false}
        id="import"
        name="import"
        onChange={[Function]}
        onClick={null}
        size={null}
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <div
      className="flex-75 layout-row layout-align-space-around-center charges_row_padding"
    >
      <p
        className="flex-none"
      >
        Export Fees
      </p>
      <Checkbox
        checked={false}
        className=""
        disabled={false}
        id="export"
        name="export"
        onChange={[Function]}
        onClick={null}
        size={null}
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
  </div>
  <div
    className="flex-100 layout-row layout-align-end-center"
  />
</div>
`;

exports[`theme is falsy 1`] = `""`;
